src/components/user.js

```diff
@@ -15,69 +21,142 @@ export default class User extends React.Component {
+            return { nameValid, phoneValid, emailValid };
+        }
+    }
+
+    _handleClickAddress (e) {
         this.setState({ showAddress: !this.state.showAddress });
         e.preventDefault();
     }
 
-    handleClickCompany (e) {
+    _handleClickCompany (e) {
         this.setState({ showCompany: !this.state.showCompany });
         e.preventDefault();
     }
 
-    handleClickEdit (e) {
+    _handleClickEdit (e) {
         const { currentUser, wantEdit } = this.state;
 
         if (wantEdit) {
             const newUser = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
 
             UserAction.updateUser(newUser);
         }
-
         this.setState({ wantEdit: !wantEdit });
         e.preventDefault();
     }
 
-    handleChange (e, type) {
-        const { currentUser, wantEdit } = this.state;
-        const {isNewUser} = this.props;
+    _handleChange (e, obj) {
+        const { currentUser, wantEdit, formErrors } = this.state;
+        const { isNewUser }                         = this.props;
 
         if (wantEdit || isNewUser) {
-            currentUser[type][e.target.name] = e.target.value;
-            this.setState({ currentUser });
+            const name  = e.target.name;
+            const value = e.target.value;
+
+            currentUser[obj][name] = value;
+            formErrors[name]       = this._validateField(name, value);
+            this.setState({ currentUser, formErrors });
         }
         e.preventDefault();
     }
 
-    handleClickSubmit () {
-        const { currentUser } = this.state;
-        const newUser         = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
+    _addUser (userId) {
+        const { isNewUser }                = this.props;
+        const { formErrors, addDuplicate } = this.state;
 
-        UserAction.addNewUser(newUser);
+        if (isNewUser && userId > 0) {
+            formErrors.duplicate[0] = 'User with this name exists. Click Submit if you want to add anyway.';
+            formErrors.duplicate[1] = `#${userId}`;
+            this.setState({ formErrors: formErrors, addDuplicate: !addDuplicate });
+        }
+    };
+
+    _handleClickSubmit (e) {
+        const { currentUser, addDuplicate, formErrors } = this.state;
+        const newUser                                   = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
+        const resultValid                               = this._validateFields(currentUser.info);
+
+        if (resultValid === true) {
+            UserAction.addNewUser(newUser, addDuplicate);
+        }
+        else {
+            formErrors.name  = resultValid.nameValid;
+            formErrors.phone = resultValid.phoneValid;
+            formErrors.email = resultValid.emailValid;
+            this.setState({ formErrors });
+        }
+        e.preventDefault();
+    }
+
+    componentDidMount () {
+        UserStore.on(EVENT_TYPE.addNewUser, this._addUser);
```

Comment:
We can subsribe for this event in each of our users
React show us a warning
![image](https://user-images.githubusercontent.com/12034551/96140275-6c88ed00-0f08-11eb-8e65-692304ffb624.png)
We have to move our subscription and pass required information to a user via props